pkg/patch/patch.go

```diff
@@ -285,13 +285,31 @@ func validateDocument(doc document.Document) error {
 	return nil
 }
 
-func validatePatches(patches []byte) error {
-	_, err := jsonpatch.DecodePatch(patches)
+func validateJSONPatches(patches []byte) error {
+	jsonPatches, err := jsonpatch.DecodePatch(patches)
 	if err != nil {
-		return err
+		return fmt.Errorf("%s: %s", JSONPatch, err.Error())
 	}
 
-	// TODO: We should probably not allow updating keys and services using this patch #175
+	for _, p := range jsonPatches {
+		pathMsg, ok := p["path"]
+		if !ok {
+			return fmt.Errorf("%s: path not found", JSONPatch)
+		}
+
+		var path string
+		if err := json.Unmarshal(*pathMsg, &path); err != nil {
+			return fmt.Errorf("%s: invalid path", JSONPatch)
+		}
+
+		if strings.HasPrefix(path, "/"+document.ServiceProperty) {
+			return fmt.Errorf("%s: cannot modify services", JSONPatch)
+		}
+
+		if strings.HasPrefix(path, "/"+document.PublicKeyProperty) {
+			return fmt.Errorf("%s: cannot modify public keys", JSONPatch)
+		}
```

Comment:
do we also need to prevent modification to root path? i.e., could someone replace the entire document at `/`?

Reply:
I will double check - I was not able to do it couple of days ago when I was working on replace patch.